### ISSUE TITLE
Figure out correct toolchain usage

### ISSUE THREAD

**Issue by Yourens** [occlum/libos#48]
Make sure it can run busybox, bash, lighttpd and gcc on occlum

**Comment by Yourens** [comment]
Expect results:  
A good musl can pass all occlum test.  
A ReadMe that introduce how to use occlum-musl build large project(like bash)

**Comment by Yourens** [comment]
怎么使用occlum 工具链编译大型项目
occlum wapper 的参数详解：

Occlum wapper contains two scripts, musl-clang and ld.musl-clang. The scripts first parses the arguments you pass to it and replaces them with those that musl and clang recognizes and can uses.
The clang is then executed by exec, giving the clang parameters at execution time to use only the musl libraries and header files. The use of musl-clang as a linker (set by fuse-ld ) is specified in musl-clang. So another script, ld.musl-clang, is called later as the linker.
Instead of using lld in ld.musl-clang, we continue to use ld because : 1. So plt is only used for syscall, and the plt for syscall is statically written.2. lld does not replace the got access to those variable have known location with the normal access, this prevents occlum from running the lld compiled binary.
In musl's wapper we do not specify binary if -FPIC and pie are compiled. Because programs compiled using the toolchain may include multiple stages, pic compilation is not required for each stage.


How to use occlum toolchain to build large project
Because the binary compiled by occlum toolchain cannot run on native, we are actually cross-compiling.
Most binaries on the linux platform are configured using autoconf or part of autoconf. So we show how occlum compiles large programs by showing how enable cross-compilation occurs in autoconf
For files generated during autoconf configuration, see the following links :
https://www-user.tu-chemnitz.de/~dker/autotools_tutorial.pdf
Autoconf has three parameter options :
--build=the architecture of the build machine
--host=the architecture that you want the file to run on
--target=the architecture that the tools target for(for gcc)

Let's take the compilation of gcc as an example:
The “configure” files generated by autoconf usually come with a config.sub to determine what platform your tools are running, and “configure" gives the corresponding compiler name based on the this platform. But changing the config.sub to add platform is too complex, so we set the cross_compiler variable to yes at “configure” directly so that “configure” run as build! = host. This is because when host=build is done, configure checks to see if the generated binary can run. But when cross_compile is yes, this check will not be performed. We then set all of build's tools to native in Makefile.in. In this way, occlum's toolchain is not used in gcc to compile commands that should be executed on the native platform.
